_archive/blech_run_auto_process.py

```diff
@@ -0,0 +1,369 @@
+# Open the hdf5 file
+if repacked_bool:
+    hdf5_name = hdf5_name[:-3] + '_repacked.h5'
+hf5 = tables.open_file(hdf5_name, 'r+')
+
+##############################
+# Instantiate unit_descriptor_handler
+this_descriptor_handler = post_utils.unit_descriptor_handler(hf5, dir_name)
+
+# Make the sorted_units group in the hdf5 file if it doesn't already exist
+if not '/sorted_units' in hf5:
+    hf5.create_group('/', 'sorted_units')
+
+autosort_output_dir = os.path.join(dir_name, 'autosort_output')
+if not os.path.isdir(autosort_output_dir):
+	os.mkdir(autosort_output_dir)
+
+############################################################
+# Main Processing Loop 
+############################################################
+# Run an infinite loop as long as the user wants to 
+# pick clusters from the electrodes   
+
+electrode_list = os.listdir('./spike_waveforms/')
+electrode_num_list = [int(re.findall(r'\d+', this_electrode)[0]) \
+		for this_electrode in electrode_list]
+electrode_num_list.sort()
+
+for electrode_num in electrode_num_list:
+
+	# electrode_output_dir = os.path.join(autosort_output_dir, f'electrode{electrode_num:02}')
+
+	############################################################
+	# Get unit details and load data
+	############################################################
+
+	print()
+	print('======================================')
+	print()
+
+	# Iterate over electrodes and pull out spikes
+	# Get classifier probabilities for each spike and use only
+	# "good" spikes
+
+	# Print out selections
+	print(f'=== Processing Electrode {electrode_num:02} ===')
+
+
+	# Load data from the chosen electrode and solution
+	# We can simply load data from 2-clusters because we're not using
+	# gmm-predictions
+	(
+		spike_waveforms,
+		spike_times,
+		pca_slices,
+		energy,
+		amplitudes,
+		predictions,
+		) = post_utils.load_data_from_disk(electrode_num, 2)
+
+	clf_data_paths = [
+			f'./spike_waveforms/electrode{electrode_num:02}/clf_prob.npy',
+			f'./spike_waveforms/electrode{electrode_num:02}/clf_pred.npy',
+			]
+	clf_prob, clf_pred = [np.load(this_path) for this_path in clf_data_paths]
+
+	# Toss all waveforms and times that are not predicted
+	spike_waveforms = spike_waveforms[clf_pred]
+	spike_times = spike_times[clf_pred]
+	pca_slices = pca_slices[clf_pred]
+	energy = energy[clf_pred]
+	amplitudes = amplitudes[clf_pred]
+	clf_prob = clf_prob[clf_pred]
+	clf_pred = clf_pred[clf_pred]
+
+	# Take everything
+	data = post_utils.prepare_data(
+			np.arange(len(spike_waveforms)),
+			pca_slices,
+			energy,
+			amplitudes,
+			)
+
+	############################################################
+	# Perform clustering
+	############################################################
+
+	# Since the dirichlet process WILL find more clusters for more data,
+	# (whereas we know that more spikes doens't mean more neurons),
+	# we need to keep the "COUNT" constant
+	# One way to do this is to use KMeans centroids as data points
+
+	max_k = 1000
```

Comment:
Maybe add to a params file in future?

Reply:
Hmm...it certainly could be added as a parameter, but I don't expect this to be needed to be changed.
I expect that if someone is digging into the algorithm so much, they'll find it anyway.
What do you think?